__tests__/integration/atomic_swap.test.js

```diff
@@ -0,0 +1,82 @@
+import { GenesisWalletHelper } from './helpers/genesis-wallet.helper';
+import {
+  createTokenHelper,
+  DEFAULT_PIN_CODE,
+  generateWalletHelper,
+  stopAllWallets,
+  waitForTxReceived,
+} from './helpers/wallet.helper';
+import { HATHOR_TOKEN_CONFIG } from '../../src/constants';
+import SendTransaction from '../../src/new/sendTransaction';
+import PartialTxProposal from '../../src/wallet/partialTxProposal';
+import storage from '../../src/storage';
+
+describe('partial tx proposal', () => {
+  afterEach(async () => {
+    await stopAllWallets();
+    await GenesisWalletHelper.clearListeners();
+  });
+
+  it('Should exchange tokens between wallets', async () => {
+    // Create the wallet
+    const hWallet1 = await generateWalletHelper();
+    const hWallet2 = await generateWalletHelper();
+    const network = hWallet1.getNetworkObject();
+
+    // Injecting funds and creating a new custom token
+    await GenesisWalletHelper.injectFunds(hWallet1.getAddressAtIndex(0), 101);
+    const { hash: token1Uid } = await createTokenHelper(
+      hWallet1,
+      'Token1',
+      'TK1',
+      100,
+    );
+
+    // Injecting funds and creating a new custom token
+    await GenesisWalletHelper.injectFunds(hWallet2.getAddressAtIndex(0), 10);
+    const { hash: token2Uid } = await createTokenHelper(
+      hWallet2,
+      'Token2',
+      'TK2',
+      1000,
+    );
+
+    /**
+     * The exchange will be:
+     *
+     * Wallet1 will send 100 HTR and 100 TK1
+     * Wallet2 will send 1000 TK2
+     */
+    const proposal = new PartialTxProposal(network);
+    // Wallet1 side
+    proposal.addSend(hWallet1, HATHOR_TOKEN_CONFIG.uid, 100);
+    proposal.addSend(hWallet1, token1Uid, 100);
+    proposal.addReceive(hWallet1, token2Uid, 1000);
+    expect(proposal.partialTx.isComplete()).toBeFalsy();
+    // Wallet2 side
+    proposal.addSend(hWallet2, token2Uid, 1000);
+    proposal.addReceive(hWallet2, HATHOR_TOKEN_CONFIG.uid, 100);
+    proposal.addReceive(hWallet2, token1Uid, 100);
+    expect(proposal.partialTx.isComplete()).toBeTruthy();
+
+    const serialized = proposal.partialTx.serialize();
+    const proposal1 = PartialTxProposal.fromPartialTx(serialized, network);
+    storage.setStore(hWallet1.store);
+    await proposal1.signData(DEFAULT_PIN_CODE, true);
+    expect(proposal1.signatures.isComplete()).toBeFalsy();
+
+    const proposal2 = PartialTxProposal.fromPartialTx(serialized, network);
+    storage.setStore(hWallet2.store);
+    await proposal2.signData(DEFAULT_PIN_CODE, true);
+
+    expect(proposal2.signatures.isComplete()).toBeFalsy();
+
+    proposal2.signatures.addSignatures(proposal1.signatures.serialize());
+    expect(proposal2.signatures.isComplete()).toBeTruthy();
+
+    const transaction = proposal2.prepareTx();
+    const sendTransaction = new SendTransaction({ transaction, network });
+    const tx = await sendTransaction.runFromMining();
+    expect(tx.hash).toBeDefined();
```

Comment:
Since this is an integration test, I would suggest some validations that the results on the wallets are the expected ones.

This may be a simple `getBalance()` on the three tokens involved, to confirm all of them are in the correct wallets, or a `getUtxos()` to validate each output.

Reply:
Made a balance check for all tokens on both wallets